packages/core/src/chain.ts

```diff
@@ -71,11 +75,28 @@ async function setChain(options: {
       code === ProviderRpcErrorCode.UNRECOGNIZED_CHAIN_ID
     ) {
       // chain has not been added to wallet
+      if (rpcUrl || label || token) {
+        if (rpcUrl) {
+          chain.rpcUrl = rpcUrl
+        }
+  
+        if (label) {
+          chain.label = label
+        }
+        
+        if (token) {
+          chain.token = token
+        }
+              
+        updateChain(chain)
+      }
+
+      // add chain to wallet
       return chainNotInWallet(
         wallet,
         chain,
         switchChainModalClosed$,
-        chainIdHex
+        chainIdHex,
```

Comment:
Has this file been prettier-ified? Trailing comma and line 27 looks a little long. No worries if it has just checking 😄